### PR TITLE
[batch2] add service account functionality

### PR DESCRIPTION
**Pull request by jigold**
I didn't know if we should have the k8s server URL in the repo. I left it out for now just in case. I haven't tested the service account changes yet.

Stacked on #7434

### CODE REVIEW

File: batch2/deployment.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ spec:
            value: "{{ batch2_worker_image.image }}"
          - name: BATCH_NAMESPACE
            value: "{{ default_ns.name }}"
+         - name: KUBERNETES_SERVER_URL
+           value: "{{ global.k8s_server_url }}"
```
Comment (cseed) [inline batch2/deployment.yaml]: This needs to be set by CI.  You should PR just adding this to CI, then this will work.

File: batch2/batch/batch.py [hunk 2]
```diff
@@ -227,6 +229,58 @@ def getopt(obj, attr):
...
+kind: Config
+preferences: {{}}
+users:
+- name: {service_account_name}
+  user:
+    token: {token}
+'''
+
+        job_spec['secrets'].append({
+            'namespace': 'batch-pods',  # FIXME unused
```
Comment (cseed) [inline batch2/batch/batch.py]: I'd just leave this out, the worker doesn't need it.

File: batch2/batch/batch.py [hunk 3]
```diff
@@ -227,6 +229,58 @@ def getopt(obj, attr):
...
+            'name': 'kube-config',
+            'mount_path': '/.kube',
+            'mount_in_copy': False,
```
Comment (cseed) [inline batch2/batch/batch.py]: This is the default.

 >                 if secret.get('mount_in_copy', False):

File: batch2/batch/front_end/front_end.py [hunk 4]
```diff
@@ -282,6 +282,11 @@
                     'mount_in_copy': True
                 })
 
+                env = spec.get('env')
+                if not env:
+                    env = []
+                    spec['env'] = env
```
Comment (cseed) [inline batch2/batch/front_end/front_end.py]: This seems unnecessary because elsewhere you still allow env to be empty.

File: batch2/batch/worker.py [hunk 5]
```diff
@@ -394,7 +402,7 @@ def __init__(self, batch_id, user, gsa_key, job_spec):
             'command': job_spec['command'],
             'image': job_spec['image'],
             'name': 'main',
-            # FIXME env
+            'env': env,
```
Comment (cseed) [inline batch2/batch/worker.py]: Thanks!

File: ci/Dockerfile.ci-utils [hunk 6]
```diff
@@ -1,7 +1,20 @@
 FROM {{ base_image.image }}
 
+# Installing kubectl on Ubuntu
+# https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-using-native-package-management
```
Comment (cseed) [inline ci/Dockerfile.ci-utils]: Why this change?  The base image already has kubectl.

**Comment by jigold** [comment]
Just for reference, this PR also moves copying the worker logs to the shutdown script and adds support for env in batch2. I also added a service account `ci-agent` in the default namespace for the test, dev scopes.

File: ci/Dockerfile.ci-utils (hunk 6 above)
Comment (jigold) [reply ci/Dockerfile.ci-utils]: I didn't see the kubectl in the base image. Fixed.

File: batch2/batch/front_end/front_end.py (hunk 4 above)
Comment (cseed) [reply batch2/batch/front_end/front_end.py]: You didn't want to get rid of htis?
